src/test/java/com/switchfully/parkshark_2024_10/user/UserRepositoryTest.java

```diff
@@ -0,0 +1,82 @@
+package com.switchfully.parkshark_2024_10.user;
+
+import com.switchfully.parkshark_2024_10.division.Division;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.context.jdbc.Sql;
+
+@DataJpaTest
+public class UserRepositoryTest {
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Test
+    void findByEmailAndPassword_shouldReturnCorrectPerson() {
+        // given
+        Division division = new Division();
+
+        Director director = new Director(division, "Jane", "Doe", "jane.doe@parkshark.com", "password123");
+        userRepository.save(director);
+
+        // When
+        Person found = userRepository.findByEmailAndPassword("jane.doe@parkshark.com", "password123");
+
+        //then
+        Assertions.assertNotNull(found, "Person should not be null");
+        Assertions.assertEquals(director.getId(), found.getId(), "Person IDs should match");
+        Assertions.assertEquals("Jane", found.getFirst_name(), "First names should match");
+        Assertions.assertEquals("Doe", found.getLast_name(), "Last names should match");
+        Assertions.assertEquals("jane.doe@parkshark.com", found.getEmail(), "Emails should match");
+    }
+
+    @Test
+    void findByEmailAndPassword_shouldReturnNullWhenCredentialsDoNotMatch() {
+        // when
+        Division division = new Division();
+
+        Director director = new Director(division, "John", "Smith", "john.smith@parjshark.com", "securePass");
+        userRepository.save(director);
+
+        // when
+        Person found = userRepository.findByEmailAndPassword("john.smith@parjshark.com", "wrongPass");
+
+        // then
+        Assertions.assertNull(found, "Person should be null when credentials do not match");
+    }
+
+    @Test
+    void findByName_shouldReturnCorrectPerson() {
```

Comment:
This only tests the firstname, not the lastname and/or a combination of both. The query that was added to the 'UserRepository.findByEmailAndPassword' method can handle partial first and lastnames as well as a 'full name' as well. This should be reflected in the tests as well.

Reply:
Everything works! Stop annoying please!